### PR TITLE
[POAE7-2434] Migrate substrait-cpp of function lookup

### PR DESCRIPTION
**Pull request by haiweizh**
### What changes were proposed in this pull request?

Migrate substrait-cpp of function lookup, use [substrait-cpp].

### Why are the changes needed?

Migrate substrait-cpp of function lookup.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Uts.

### Which label does this PR belong to?

CODEREFACTOR,cider,test

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,6 +43,7 @@ option(CIDER_ENABLE_VELOX "Build Velox Plugin for Cider." ON)
 option(CIDER_ENABLE_SUBSTRAIT "Build Substrait for Cider." ON)
 option(CIDER_ENABLE_BENCHMARK "Build benchmark for Cider." ON)
 option(CIDER_ENABLE_YAMLCPP "Build yaml-cpp for Cider." ON)
+option(CIDER_ENABLE_SUBSTRAIT_CPP "Build substrait-cpp for Cider." ON)
```
Comment (yma11) [inline CMakeLists.txt:46]: My understanding is that this option is always `ON`, right?

**Review by yma11 — DISMISSED** [review]
Overall LGTM.

**Comment by haiweizh** [comment]
Compilation conflicts, need to been updated.

**Review by chaojun-zhang — DISMISSED** [review]
LGTM

**Comment by haiweizh** [comment]
Compilation conflicts, need to been updated. Don't merge into main branch.

**Pull request closed without merging**